pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -821,6 +821,8 @@ type PodGC struct {
 	Strategy PodGCStrategy `json:"strategy,omitempty" protobuf:"bytes,1,opt,name=strategy,casttype=PodGCStrategy"`
 	// LabelSelector is the label selector to check if the pods match the labels before being added to the pod GC queue.
 	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty" protobuf:"bytes,2,opt,name=labelSelector"`
+	// DeleteDelaySeconds is how long to delay pod deletion for.
+	DeleteDelaySeconds uint32 `json:"deleteDelaySeconds,omitempty" protobuf:"bytes,3,opt,name=deleteDelaySeconds"`
```

Comment:
I think this should be a general GC delay.

Reply:
If we do that I assume we want to delay all GC actions including things like labelling pods? in my case I only care about delaying the deletion, I cannot think of a scenario where I would want to delay labelling the pod as complete.

Reply:
This only applies to `PodGCStrategy` - labeling is not really part of that. I think `DelaySeconds` would be good unless @jessesuen should we prefer `time.Duration` to `*Seconds`?

Reply:
Thanks @alexec have changed to `DelaySeconds` will await further feedback from @jessesuen on if we prefer a Duration or not.

Reply:
@alexc @jessesuen any more feedback on this PR?

Reply:
@sgloutnikov  @alexec Heads up that I keep getting quite a lot of incorrect mentions from this repository from you guys .... it appears we have very similar usernames

Reply:
> @sgloutnikov  @alexec Heads up that I keep getting quite a lot of incorrect mentions from this repository from you guys .... it appears we have very similar usernames

Oops sorry my bad!